crates/fj-host/src/host_handle.rs

```diff
@@ -0,0 +1,82 @@
+use std::thread::JoinHandle;
+
+use crossbeam_channel::Sender;
+
+use crate::{EventLoopClosed, Model};
+
+/// A handle for sending commands to a spawned host
+pub struct HostHandle {
+    command_tx: Sender<HostCommand>,
+    host_thread: Option<JoinHandle<Result<(), EventLoopClosed>>>,
+    model_loaded: bool,
+}
+
+impl HostHandle {
+    /// Create a `HostHandle` with a send channel and the host thread handle.
+    pub fn new(
+        command_tx: Sender<HostCommand>,
+        host_thread: JoinHandle<Result<(), EventLoopClosed>>,
+    ) -> Self {
+        Self {
+            command_tx,
+            host_thread: Some(host_thread),
+            model_loaded: false,
+        }
+    }
+
+    /// Send a model to the host for evluation.
+    pub fn load_model(&mut self, model: Model) {
+        self.command_tx
+            .try_send(HostCommand::LoadModel(model))
+            .expect("Host channel disconnected unexpectedly");
+        self.model_loaded = true;
+    }
+
+    /// Whether a model has been sent to a host for watching and evaluation
+    pub fn is_model_loaded(&self) -> bool {
+        self.model_loaded
+    }
+
+    /// Check if the host thread has exited with a panic. This method runs at
+    /// each tick of the event loop. Without an explicit check, an operation
+    /// will appear to hang forever (e.g. processing a model).  An error
+    /// will be printed to the terminal, but the gui will not notice until
+    /// a new `HostCommand` is issued on the disconnected channel.
```

Comment:
I think this is fine, but I'd like to mention a possible alternative: We could wrap the the host thread in [`catch_unwind`](https://doc.rust-lang.org/std/panic/fn.catch_unwind.html) and send an error message through the channel, if it panics.

Reply:
That's a good idea and would probably be a better API as it would remove the need to explicitly poll for panics. 

I did it this way to keep the existing behavior intact when there is a panic in `ShapeProcessor`.  It crashes `fj-app` completely when the shape processing is in the main thread. My thinking was that a panic is unrecoverable, but I can see how that's not always the case .  In this case the user could always edit the model and try again, if we built that kind of feature in.

Note that if the host thread crashes, the app would be in an unusable state as things are currently.  An error message could explain the situation, but might be confusing for the user. Maybe we should leave things as-is for now, and revisit error handling as a separate issue?